### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by the-sashko**

### CODE REVIEW

File: public/assets/css/app.css [hunk 1]
```diff
@@ -1,691 +1,109 @@
...
+    width:       100%;
+    text-align:  center;
+    font-weight: normal;
 }
-.pagination a:active{
-    background: #060;
-    color: #FEFEFE;
+
+a {
+    text-decoration: none;
```
Comment (diogenes-33) [inline public/assets/css/app.css:68]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

**Comment by the-sashko** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 539
- Added 252
           

Complexity increasing per file
==============================
- protected/models/moderator/moderator.object.php  1
- protected/models/auth/auth.vo.php  1
- public/assets_old/js/fastclick.js  24
- protected/models/board/board.vo.php  1
- protected/models/media/media.php  1
- protected/models/media/media.object.php  1
- protected/models/session/session.php  1
- protected/models/board/board.php  1
- protected/models/auth/auth.object.php  1
- protected/models/session/session.object.php  1
- protected/models/board/board.object.php  1
- protected/models/moderator/moderator.vo.php  1
- protected/models/moderator/moderator.php  1
- protected/models/media/media.vo.php  1
         

Complexity decreasing per file
==============================
+ protected/models/user/user.php  -3
+ protected/models/post/post.vo.php  -2
+ protected/models/post/post.php  -10
+ protected/controllers/UserController.php  -2
+ protected/models/session/session.vo.php  -2
+ protected/models/user/user.vo.php  -1
+ protected/controllers/PostController.php  -4
+ protected/controllers/CronController.php  -3
+ protected/controllers/AjaxController.php  -3
+ protected/controllers/AdminController.php  -4
+ protected/controllers/MainController.php  -1
         

Clones added
============
- public/assets_old/js/fastclick.js  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482&bid=18989671)

File: public/assets_old/css/admin.css [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	width:100%;
+	background: #48A;
+}
+.header h1 {
+	text-align: center;
+	font-size: 1.25em;
+	font-weight: bold;
+	color: #FEFEFE;
+	text-transform: uppercase;
+	padding: .25em 0;
```
Comment (diogenes-33) [inline public/assets_old/css/admin.css:35]: Codacy found an issue: [Expected a leading zero (number-leading-zero)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/routers/cli.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ * Trait For Routing HTTP Requests
+ */
+trait Router
+{
+    /**
+     * Perform Redirect Rules
+     *
+     * @param string $uri HTTP Request URI
+     */
+    public function routeRedirect(?string $uri = null): void
```
Comment (diogenes-33) [inline protected/routers/cli.php:12]: Codacy found an issue: [TRUE, FALSE and NULL must be uppercase; expected "NULL" but found "null"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/js/fastclick.js [hunk 4]
```diff
@@ -0,0 +1,841 @@
...
+			break;
+		case 'input':
+
+			// File inputs need real clicks on iOS 6 due to a browser bug (issue #68)
+			if ((deviceIsIOS && target.type === 'file') || target.disabled) {
+				return true;
+			}
+
+			break;
+		case 'label':
```
Comment (diogenes-33) [inline public/assets_old/js/fastclick.js:247]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/css/admin.css [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+	font-family: Arial,sans-serif;
+	font-size: 16px;
+	color:#333;
+	background:#FEFEFE;
+	min-width:320px;
+}
+input {
+	font-size: 1.125em;
+	padding: .125em .25em;
+	border: 1px solid #6A6A6A;
```
Comment (diogenes-33) [inline public/assets_old/css/admin.css:15]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/js/fastclick.js [hunk 6]
```diff
@@ -0,0 +1,841 @@
...
+		case 'iframe': // iOS8 homescreen apps can prevent events bubbling into frames
+		case 'video':
```
Comment (diogenes-33) [inline public/assets_old/js/fastclick.js:249]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/cron/cron.vo.php [hunk 7]
```diff
@@ -107,94 +58,53 @@ public function getStatus() : string
...
 
         if (strlen($errorMessage) > 255) {
-            $errorMessage = mb_substr($errorMessage, 0, 254).'…';
+            $errorMessage = mb_substr($errorMessage, 0, 254);
+            $errorMessage = sprintf('%s…', $errorMessage);
         }
 
-        return $this->set('error_message', $errorMessage);
+        $this->set('error_message', $errorMessage);
+        return true;
```
Comment (diogenes-33) [inline protected/models/cron/cron.vo.php:108]: Codacy found an issue: [TRUE, FALSE and NULL must be uppercase; expected "TRUE" but found "true"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/js/fastclick.js [hunk 8]
```diff
@@ -0,0 +1,841 @@
...
+		case 'textarea':
+			return true;
+		case 'select':
+			return !deviceIsAndroid;
+		case 'input':
+			switch (target.type) {
+			case 'button':
+			case 'checkbox':
+			case 'file':
+			case 'image':
```
Comment (diogenes-33) [inline public/assets_old/js/fastclick.js:274]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/media/media.vo.php [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+ * ValuesObject Class For Example Model
+ */
+class ExampleVO extends ValuesObject
+{
+    /**
+     * Get Example ID
+     *
+     * @return int Example ID
+     */
+    public function getID(): int
```
Comment (diogenes-33) [inline protected/models/media/media.vo.php:12]: Codacy found an issue: [Public method name "ExampleVO::getID" is not in camel caps format](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets/css/app.css [hunk 10]
```diff
@@ -1,691 +1,109 @@
...
 }
-.pagination a{
-    display: inline;
-    background: #080;
-    color: #FEFEFE;
-    margin: 0 .25em;
-    padding: .25em .375em;
+
+h5 {
+    font-size:   1.325em;
```
Comment (diogenes-33) [inline public/assets/css/app.css:52]: Codacy found an issue: [Expected single space after ":" with a single-line declaration (declaration-colon-space-after)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/cron/cron.object.php [hunk 11]
```diff
@@ -38,42 +20,31 @@ public function create(array $values = []) : bool
...
-     *
-     * @return bool Is Cron Job Data Successfully Updated
+     * summary
      */
     public function updateCronByID(
-        array  $columns = [],
-        array  $values  = [],
-        int    $id      = -1
-    ) : bool
+        ?array  $columns = null,
```
Comment (diogenes-33) [inline protected/models/cron/cron.object.php:26]: Codacy found an issue: [TRUE, FALSE and NULL must be uppercase; expected "NULL" but found "null"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/cron/cron.object.php [hunk 12]
```diff
@@ -1,32 +1,14 @@
...
-/**
- * ModelObject Class For Cron Model
- */
 class CronObject extends ModelObjectCore
 {
-    /**
-     * @var string Default Data Base Table
-     */
     public $defaultTableName = 'cron_jobs';
+    public $tableCrons       = 'cron_jobs';
```
Comment (diogenes-33) [inline protected/models/cron/cron.object.php:5]: Codacy found an issue: [Missing member variable doc comment](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/cron/cron.php [hunk 13]
```diff
@@ -1,65 +1,35 @@
...
-     *
-     * @return bool Is Cron Job Exists
-     */
-    private function _isCronExists(
-        string $action   = '',
-        int    $interval = -1
-    ) : bool
+    protected function _isCronExists(
+        ?string $action   = null,
+        ?int    $interval = null
```
Comment (diogenes-33) [inline protected/models/cron/cron.php:20]: Codacy found an issue: [Avoid unused parameters such as '$interval'.](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/css/admin.css [hunk 14]
```diff
@@ -0,0 +1,159 @@
...
+	border-radius: .125em;
+	min-width: 300px;
+}
+form{
```
Comment (diogenes-33) [inline public/assets_old/css/admin.css:19]: Codacy found an issue: [Expected single space before "{" (block-opening-brace-space-before)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/cron/cron.vo.php [hunk 15]
```diff
@@ -107,94 +58,53 @@ public function getStatus() : string
...
         }
 
         if (strlen($errorMessage) > 255) {
-            $errorMessage = mb_substr($errorMessage, 0, 254).'…';
+            $errorMessage = mb_substr($errorMessage, 0, 254);
+            $errorMessage = sprintf('%s…', $errorMessage);
         }
 
-        return $this->set('error_message', $errorMessage);
+        $this->set('error_message', $errorMessage);
```
Comment (diogenes-33) [inline protected/models/cron/cron.vo.php:107]: Codacy found an issue: [Variable "$errorMessage" is not in valid snake_case format, try "$error_message"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/cron/cron.form.php [hunk 16]
```diff
@@ -84,46 +57,36 @@ private function _getFormFields(
...
-     */
-    private function _validateFormFields(
-        string $action   = '',
-        int    $interval = -1,
-        bool   $isActive = FALSE
-    ) : array
+    protected function _validateFormFields(
+        ?string $action   = null,
+        ?int    $interval = null,
+        bool    $isActive = false
```
Comment (diogenes-33) [inline protected/models/cron/cron.form.php:63]: Codacy found an issue: [Variable "$isActive" is not in valid snake_case format, try "$is_active"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/board/board.vo.php [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+ * ValuesObject Class For Example Model
+ */
+class ExampleVO extends ValuesObject
+{
+    /**
+     * Get Example ID
+     *
+     * @return int Example ID
+     */
+    public function getID(): int
```
Comment (diogenes-33) [inline protected/models/board/board.vo.php:12]: Codacy found an issue: [Public method name "ExampleVO::getID" is not in camel caps format](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/controllers/PostController.php [hunk 18]
```diff
@@ -1,94 +1,36 @@
...
-        list($status, $errors) = $this->getModel('post')->create($this->post);
-
-        if ($status) {
-            $user->setFlashSuccess();
-            $this->redirect($user->_getRedirectURL());
-        }
+    /**
+     * @var string Template Scope
+     */
+    public $templaterScope = 'site';
```
Comment (diogenes-33) [inline protected/controllers/PostController.php:10]: Codacy found an issue: [Member variable "$templaterScope" is not in valid snake_case format, try "$templater_scope"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/models/moderator/moderator.object.php [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /**
+     * @var string Default Data Base Table
+     */
+    public $defaultTableName = 'example';
+
+    /**
+     * @var string Data Base Table For Example Data
+     */
+    public $tableExample = 'example';
```
Comment (diogenes-33) [inline protected/models/moderator/moderator.object.php:15]: Codacy found an issue: [Member variable "$tableExample" is not in valid snake_case format, try "$table_example"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets/css/app.css [hunk 20]
```diff
@@ -1,691 +1,109 @@
...
-.button {
-    margin: 0 !important;
-    padding: .5em !important;
-    background-color: #080 !important;
-    font-size: 0.825em !important;
-    text-transform: uppercase !important;
-    font-weight: bold !important;
-    border: 0 !important;
-    border-radius: 0 !important;
+    text-align:  center;
```
Comment (diogenes-33) [inline public/assets/css/app.css:13]: Codacy found an issue: [Expected single space after ":" with a single-line declaration (declaration-colon-space-after)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: protected/routers/cli.php [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    /**
+     * Perform Rewrite Rules
+     *
+     * @param string $uri HTTP Request URI
+     *
+     * @return string Rewrited HTTP Request URI
+     */
+    public function routeRewrite(?string $uri = null): string
```
Comment (diogenes-33) [inline protected/routers/cli.php:31]: Codacy found an issue: [Method name "routeRewrite" in class Router is not in snake case format, try "route_rewrite"](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/js/fastclick.js [hunk 22]
```diff
@@ -0,0 +1,841 @@
...
+		case 'input':
+
+			// File inputs need real clicks on iOS 6 due to a browser bug (issue #68)
+			if ((deviceIsIOS && target.type === 'file') || target.disabled) {
+				return true;
+			}
+
+			break;
+		case 'label':
+		case 'iframe': // iOS8 homescreen apps can prevent events bubbling into frames
```
Comment (diogenes-33) [inline public/assets_old/js/fastclick.js:248]: Codacy found an issue: [Strings must use doublequote.](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/css/admin.css (hunk 14 above)
Comment (diogenes-33) [inline public/assets_old/css/admin.css:19]: Codacy found an issue: [Expected empty line before rule (rule-empty-line-before)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/css/admin.css [hunk 23]
```diff
@@ -0,0 +1,159 @@
...
+	font-size: 1.25em;
+	font-weight: bold;
+	color: #FEFEFE;
+	text-transform: uppercase;
+	padding: .25em 0;
+}
+.clear{
+	content:"";
+	display:block;
+	clear:both
```
Comment (diogenes-33) [inline public/assets_old/css/admin.css:40]: Codacy found an issue: [Expected single space after ":" with a single-line declaration (declaration-colon-space-after)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

File: public/assets_old/css/admin.css [hunk 24]
```diff
@@ -0,0 +1,159 @@
...
+}
+body {
+	font-family: Arial,sans-serif;
+	font-size: 16px;
+	color:#333;
+	background:#FEFEFE;
+	min-width:320px;
+}
+input {
+	font-size: 1.125em;
```
Comment (diogenes-33) [inline public/assets_old/css/admin.css:13]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/manual/common/buryak/pullRequest?prid=5800482)

**Pull request merged**